Make not updating BTCD

Just loaded commit 147596047f5fac79cba4128a6a2aa47bce2e6d7c of LND.
I ran Make, then make check.

Make check yields the following error:
github.com/btcsuite/btcd && git checkout f899737d7f2764dc13e4d01ff00108ec58f766a9 && glide install
fatal: reference is not a tree: f899737d7f2764dc13e4d01ff00108ec58f766a9
Makefile:121: recipe for target 'btcd' failed

It appears that Make isn't pulling the latest btcd.  I already had BTCD at an older commit level.

Manually going to BTCD directory and Git Pull is a workaround.

--- a/Makefile
+++ b/Makefile
@@ -97,7 +97,7 @@ $(DEP_BIN):
 $(GLIDE_BIN):
 	@$(call print, "Fetching glide.")
 	go get -d $(GLIDE_PKG)
-	cd ${GOPATH}/src/$(GLIDE_PKG) && git checkout $(GLIDE_COMMIT) 
+	cd ${GOPATH}/src/$(GLIDE_PKG) && ( git checkout $(GLIDE_COMMIT) || ( git fetch --all && git checkout $(GLIDE_COMMIT) ) )
 	$(GOINSTALL) $(GLIDE_PKG)
 
 $(GOVERALLS_BIN):
@@ -119,7 +119,7 @@ $(BTCD_DIR):
 
 btcd: $(GLIDE_BIN) $(BTCD_DIR)
 	@$(call print, "Compiling btcd dependencies.")
-	cd $(BTCD_DIR) && git checkout $(BTCD_COMMIT) && glide install
+	cd $(BTCD_DIR) && ( git checkout $(BTCD_COMMIT) || ( git fetch --all && git checkout $(BTCD_COMMIT) ) ) && glide install
 	@$(call print, "Installing btcd and btcctl.")
 	$(GOINSTALL) $(BTCD_PKG)
 	$(GOINSTALL) $(BTCD_PKG)/cmd/btcctl

